### PR TITLE
feat(auth)：add auth sdk

### PR DESCRIPTION
**Pull request by 15168440402**

### CODE REVIEW

**Comment by zhenlei520** [comment]
Please ensure that the constraints in BuildingBlocks exist before submitting the Contrib pr

File: src/BasicAbility/Masa.Contrib.BasicAbility.Auth/AuthClient.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    public IUserService UserService { get; }
+
+    public ISubjectService SubjectService { get; }
+
+    public ITeamService TeamService { get; }
+
+    public List<SubjectModel> GetList(ICallerProvider callerProvider,string requestUri,string filter)
+    {
+        var a = callerProvider.GetAsync<object, List<SubjectModel>>(requestUri, new { filter }).Result;
```
Comment (doddgu) [inline src/BasicAbility/Masa.Contrib.BasicAbility.Auth/AuthClient.cs:23]: a?

File: src/BasicAbility/Masa.Contrib.BasicAbility.Auth/AuthClient.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        TeamService = new TeamService(callerProvider);
+    }
+
+    public IUserService UserService { get; }
+
+    public ISubjectService SubjectService { get; }
+
+    public ITeamService TeamService { get; }
+
+    public List<SubjectModel> GetList(ICallerProvider callerProvider,string requestUri,string filter)
```
Comment (doddgu) [inline src/BasicAbility/Masa.Contrib.BasicAbility.Auth/AuthClient.cs:21]: code format

File: src/BasicAbility/Masa.Contrib.BasicAbility.Auth/Masa.Contrib.BasicAbility.Auth.csproj [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Masa.Utils.Caller.HttpClient" Version="0.4.0-preview.4" />
```
Comment (doddgu) [inline src/BasicAbility/Masa.Contrib.BasicAbility.Auth/Masa.Contrib.BasicAbility.Auth.csproj:10]: Use `Directory.Build.props`

File: test/Masa.Contrib.BasicAbility.Auth.Tests/Masa.Contrib.BasicAbility.Auth.Tests.csproj [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+
+    <IsPackable>false</IsPackable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.1.0" />
```
Comment (doddgu) [inline test/Masa.Contrib.BasicAbility.Auth.Tests/Masa.Contrib.BasicAbility.Auth.Tests.csproj:12]: Use `Directory.Build.props`

**Review by doddgu — CHANGES_REQUESTED** [review]

**Review by doddgu — APPROVED** [review]
LGTM

File: src/BasicAbility/Masa.Contrib.BasicAbility.Auth/AuthClient.cs (hunk 1 above)
Comment (15168440402) [reply src/BasicAbility/Masa.Contrib.BasicAbility.Auth/AuthClient.cs]: deleted

**Pull request merged**